### ISSUE TITLE
Undetected failure of generate_psa_constants.py on Windows

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls-test#7]
Observed in the logs of the `win32-msvc12_64` job on `development` (e.g. [nightly](https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-nightly-tests/1375/execution/node/2440/log/?consoleFull)):
```
C:\builds\workspace\mbed-tls-nightly-tests\src>if exist scripts\generate_psa_constants.py scripts\generate_psa_constants.py
  File "C:\builds\workspace\mbed-tls-nightly-tests\src\scripts\generate_psa_constants.py", line 294
    _nonascii_re = re.compile(rb'[^\x00-\x7f]+')
                                              ^
SyntaxError: invalid syntax
```
But the build script continues and reports success.

The failure to run `generate_psa_constants` is not a problem because calling `generate_psa_constants` is no longer needed, and it's been long enough that we probably don't have any old PR that would be blocked by removing it. However, the fact that a failure is ignored is worrying.

Goals of this task:

* Remove the call to `generate_psa_constants` (or determine that it's still needed).
* Investigate what it would take to detect failures, and make it so (or maybe add some documentation somewhere if removing the call to `generate_psa_constants` causes the script not to have any more undetected failure conditions).
* Figure out what Python version we have on Windows, and either upgrade it or document which of our scripts need to be compatible with that older version.
